eng/pipelines/templates/stages/archetype-spec-gen-sdk.yml

```diff
@@ -29,257 +29,251 @@ parameters:
+                -PRBody "$(GeneratedSDKInformation)"
+                -OpenAsDraft $true
 
-              - task: PowerShell@2
-                displayName: Create pull request
-                condition: and(succeeded(), eq(variables['HasChanges'], 'true'), ne(variables['Build.Reason'], 'PullRequest'), not(endsWith(variables['SdkRepoName'], '-pr')))
-                inputs:
-                  pwsh: true
-                  workingDirectory: $(SdkRepoDirectory)
-                  filePath: $(SdkRepoDirectory)/eng/common/scripts/Submit-PullRequest.ps1
-                  arguments: >
-                    -RepoOwner "$(SdkRepoOwner)"
-                    -RepoName "$(SdkRepoName)"
-                    -BaseBranch "main"
-                    -PROwner "azure-sdk"
-                    -PRBranch "$(PrBranch)-$(Build.BuildId)"
-                    -AuthToken "$(azuresdk-github-pat)"
-                    -PRTitle "$(PrTitle)-generated-from-$(Build.DefinitionName)-$(Build.BuildId)"
-                    -PRBody "$(GeneratedSDKInformation)"
-                    -OpenAsDraft $true
+      - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+        - pwsh: |
+            . $(SpecRepoDirectory)/eng/common/scripts/Helpers/ApiView-Helpers.ps1
+            Create-API-Review `
+              -specGenSDKArtifactPath "$(System.DefaultWorkingDirectory)/$(SpecGenSdkArtifactPath)/$(SpecGenSdkArtifactName).json" `
+              -apiviewArtifactName "$(PipelineArtifactsName)" `
+              -buildId "$(Build.BuildId)" `
+              -commitish "$(System.PullRequest.SourceCommitId)" `
+              -repoName "$(SdkRepoOwner)/$(SpecRepoName)" `
+              -pullRequestNumber "$(System.PullRequest.PullRequestNumber)" 
+          displayName: 'Detect API Change and Create API Review'
+          condition: and(succeeded(), ne(variables['SpecGenSdkArtifactPath'], ''), ne(variables['SpecGenSdkArtifactName'], ''))
 
-          - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
-            - pwsh: |
-                . $(SpecRepoDirectory)/eng/common/scripts/Helpers/ApiView-Helpers.ps1
-                Create-API-Review `
-                  -specGenSDKArtifactPath "$(System.DefaultWorkingDirectory)/$(SpecGenSdkArtifactPath)/$(SpecGenSdkArtifactName).json" `
-                  -apiviewArtifactName "$(PipelineArtifactsName)" `
-                  -buildId "$(Build.BuildId)" `
-                  -commitish "$(System.PullRequest.SourceCommitId)" `
-                  -repoName "$(SdkRepoOwner)/$(SpecRepoName)" `
-                  -pullRequestNumber "$(System.PullRequest.PullRequestNumber)" 
-              displayName: 'Detect API Change and Create API Review'
-              condition: and(succeeded(), ne(variables['SpecGenSdkArtifactPath'], ''), ne(variables['SpecGenSdkArtifactName'], ''))
\ No newline at end of file
+      - template: /eng/common/pipelines/templates/steps/publish-artifact.yml
```

Comment:
Why not keep this step along with other's artifact publish step? This step should be run even if APIView creation step fails.

Reply:
I just put this at the end like it was before but I'm fine with publishing them together given the retry doesn't currently work anyway.

Reply:
Right, retry won't work if ApiView step fails.